### DISCUSSION TITLE
User Libraries

### DISCUSSION THREAD

**Discussion by lort-Z** [ra3xdh/qucs_s#699] [Q&A] [answered]
Does somebody share libraries of components, creatate by themselves?
Is there a guide, how to create SPICE models of components, using charts from datasheets?
I would really appriciete abbilyty to use specific non linear components, but lot of them arent present in base pacage for now.

**Comment by ra3xdh** [comment, marked as answer]
> Does somebody share libraries of components, creatate by themselves?

You can propose a PR with libraries into the main repository or in this repository: https://github.com/ra3xdh/qucs-rus-complib You can find a library design guide here: https://habr.com/ru/articles/805963/ but only in Russian

> Is there a guide, how to create SPICE models of components, using charts from datasheets?

You should look for articles using keywords "SPICE model extraction" A basic guide for MOSFET could be found here: https://huskiecommons.lib.niu.edu/cgi/viewcontent.cgi?article=5043&context=allgraduate-thesesdissertations It is a complicated procedure that requires a special proprietary software like IC-CAP. The data from the datasheet may be not sufficient. It is required to measure a full set of IV-curves. 

There is no common route to design a SPICE model for more complicated ICs like opamps or another analog ICs.  Again a set of measurements is required and datasheet data may be not sufficient.
